src/app/dwg/qgsdwgimportdialog.cpp

```diff
@@ -535,3 +539,55 @@ void QgsDwgImportDialog::showHelp()
 {
   QgsHelp::openHelp( QStringLiteral( "managing_data_source/opening_data.html#importing-a-dxf-or-dwg-file" ) );
 }
+
+void QgsDwgImportDialog::layers_clicked( QTableWidgetItem *item )
+{
+  if ( ! item )
+    return;
+
+  if ( item->column() != static_cast<int>( ColumnIndex::Name ) )
+    item = mLayers->item( item->row(), static_cast<int>( ColumnIndex::Name ) );
+
+  if ( ! item )
+    return;
+
+  const QgsTemporaryCursorOverride waitCursor( Qt::WaitCursor );
+
+  QString layerName = item->text();
+
+  QgsLayerTreeGroup layerGroup;
+  createGroup( &layerGroup, layerName, QStringList( layerName ), true );
+
+  QList<QgsMapLayer *> layers = layerGroup.layerOrderRespectingGroupLayers();
+
+  if ( layers.isEmpty() )
+  {
+    mLabelLayerPreview->setText( tr( "No features in this layer" ) );
+    return;
+  }
+
+  // Create image and painterr
+  QImage image( mLabelLayerPreview->size(), QImage::Format_ARGB32_Premultiplied );
```

Comment:
Why not just use a QgsMapCanvas here? That's what the preview feature in browser uses, along with a present pan tool, and it works very nicely.

Reply:
@nyalldawson thanks, @nirvn already told me the same 🙂. Its for sure much better I'll change it!